parachain/primitives/router/src/lib.rs

```diff
@@ -110,3 +178,129 @@ where
 		}
 	}
 }
+
+pub struct EthereumSovereignAccountConvertsFor<AccountId>(PhantomData<AccountId>);
+impl<AccountId: From<[u8; 32]> + Clone> Convert<MultiLocation, AccountId>
+	for EthereumSovereignAccountConvertsFor<AccountId>
+{
+	fn convert_ref(location: impl Borrow<MultiLocation>) -> Result<AccountId, ()> {
+		log::trace!(
+			target: "xcm::location_conversion",
+			"EthereumSovereignAccountConvertsFor location: {:?}",
+			location.borrow(),
+		);
+
+		if let MultiLocation {
+			parents: 2,
+			interior: X2(GlobalConsensus(Ethereum { chain_id }), AccountKey20 { network, key }),
```

Comment:
Should disregard `parents` for GlobalConsensus locations

```suggestion
			interior: X2(GlobalConsensus(Ethereum { chain_id }), AccountKey20 { network, key }), ..
```